tests/reftests/lint.test

```diff
@@ -884,7 +885,53 @@ license: "ISC"
 dev-repo: "hg+https://to@li.nt"
 bug-reports: "https://nobug"
 url { src:"git+https://a/repo" }
-### # package with git url
+### opam lint ./lint.opam
+${BASEDIR}/lint.opam: Passed.
+### opam lint ./lint.opam --check-upstream
+${BASEDIR}/lint.opam: Passed.
+### # package with local url
+### <repo/stuff>
+something
+### <lint.opam>
+opam-version: "2.0"
+synopsis: "A word"
+description: "Two words."
+authors: "the testing team"
+homepage: "egapemoh"
+maintainer: "maint@tain.er"
+license: "ISC"
+dev-repo: "hg+https://to@li.nt"
+bug-reports: "https://nobug"
+### <add-url.sh>
+basedir=`echo "$BASEDIR" | sed 's/\\\\/\\\\\\\\\\\\\\\\/g'`
+cat << EOF >> lint.opam
+url { src:"file://$basedir/an-archive" }
+EOF
+### sh add-url.sh
+### opam lint ./lint.opam
+${BASEDIR}/lint.opam: Passed.
+### opam lint ./lint.opam --check-upstream
+${BASEDIR}/lint.opam: Passed.
+### # package with local archive url
+### <lint.opam>
+opam-version: "2.0"
+synopsis: "A word"
+description: "Two words."
+authors: "the testing team"
+homepage: "egapemoh"
+maintainer: "maint@tain.er"
+license: "ISC"
+dev-repo: "hg+https://to@li.nt"
+bug-reports: "https://nobug"
+### <add-url.sh>
+basedir=`echo "$BASEDIR" | sed 's/\\\\/\\\\\\\\\\\\\\\\/g'`
```

Comment:
you used single quotes here instead of double-quotes and only halved the left part of the sed, not the right part.
```suggestion
basedir=`echo "$BASEDIR" | sed 's/\\\\/\\\\\\\\/g'`
```

Reply:
actually what is the goal of this sed? Is it just to escape every backslashes? If so, then the solution might be instead:
```suggestion
basedir=`echo "$BASEDIR" | sed 's/\\/\\\\/g'`
```
I know is everywhere in the reftests in its double-quoted form but i’ve never really understood it ^^"

Reply:
It's the different levels of escape that are needed to end with `s/\\/\\\\/g` launched.